src/atomization/atomizer.py

```diff
@@ -337,72 +342,127 @@ def execute_delete_command(package_id: int) -> int:
+                    ),
                 )
-                atom_id = cursor.lastrowid
-                atom_id_map[atom['identifier']] = atom_id
-                
-                # If there's a parent, add the dependency relationship
-                if parent_id is not None:
-                    cursor.execute(
+                atom_id_map[identifier] = cursor.lastrowid
+
+            # Second pass: Insert new dependencies
+            for atom in atoms_to_process:
+                parent_id = atom_id_map.get(atom.get("identifier"))
+                deps = atom.get("deps")
+
+                cursor.execute(
+                    """
+                    SELECT DISTINCT childatom_id 
+                    FROM atomsdependencies 
+                    WHERE parentatom_id = %s 
+                    AND childatom_id IS NOT NULL
+                    """,
+                    (parent_id,),
+                )
+                existing_deps = {
+                    row.get("childatom_id")
+                    for row in cursor.fetchall()
+                    if row.get("childatom_id") is not None
+                }
+
+                if deps and isinstance(
+                    deps[0], str
+                ):  # Isabelle format: deps are identifiers
+                    dep_ids = [
+                        (parent_id, atom_id_map.get(dep))
+                        for dep in deps
+                        if atom_id_map.get(dep) not in existing_deps
+                    ]
+                else:  # Lean format: deps are full atoms
+                    dep_ids = [
+                        (parent_id, atom_id_map.get(dep.get("identifier")))
+                        for dep in deps
+                        if atom_id_map.get(dep.get("identifier")) not in existing_deps
+                    ]
+
+                if dep_ids:
+                    print(
+                        f"Inserting new dependencies for atom {atom.get('identifier')}: {dep_ids}"
+                    )
+                    cursor.executemany(
                         "INSERT INTO atomsdependencies (parentatom_id, childatom_id) VALUES (%s, %s)",
-                        (parent_id, atom_id)
+                        dep_ids,
                     )
-                
-                # Process dependencies recursively
-                for dep in atom.get('deps', []):
-                    process_atom(dep, atom_id)
-                
-                return atom_id
-            
-            # Process all top-level atoms
-            for atom in parsed_chunks:
-                process_atom(atom)
-            
+
             conn.commit()
-            logger.info(f"Successfully saved {len(atom_id_map)} atoms to database")
+            logger.info(f"Successfully processed {len(atom_id_map)} atoms")
             return True
-            
+
     except MysqlConnectorError as e:
         logger.error(f"Database error while saving atoms: {e}")
         return False
 
 
+def delete_all_atoms() -> bool:
```

Comment:
We need this function to take in a code_id and delete only the atoms and atomsdependencies entries associated with that code id.